### PR TITLE
Confirmations code re-structuring

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**
Restructuring **Confirmations** related code under `confirmations` domain.

## **Related issues**

Fixes: MetaMask/MetaMask-planning#1862

## **Manual testing steps**
NA

## **Screenshots/Recordings**
NA

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [X] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by digiwand — COMMENTED** [review]
Q:
should we move the other confirm-** files into confirmation? otherwise why not move them? 
- confirm-add-suggested-nft
- confirm-add-suggested-token.js
- confirm-decrypt-message
- confirm-encryption-public-key

Screenshot 2024-02-01 at 11 24 23 PM

File: ui/pages/confirmations/index.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/** Please import your files in alphabetical order **/
+@import 'confirm-approve/index';
+@import 'confirmation/confirmation';
+@import 'send/gas-display/index';
+@import 'send/send';
+@import 'token-allowance/index';
+@import './components/index'
```
Comment (digiwand) [inline ui/pages/confirmations/index.scss:7]: 👍🏼

**Review by digiwand — DISMISSED** [review]
Great restructuring work! Lgtm

I like how some components have been flattened when moved to confirmations/components e.g. modals. We can consider further restructuring in the future following the confirmations refactor project

**Review by segun — DISMISSED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,27 +3,30 @@ version: 2.1
 executors:
   node-browsers:
     docker:
-      - image: cimg/node:18.18-browsers
+      - image: cimg/node:20.11-browsers
```
Comment (digiwand) [inline .circleci/config.yml]: interesting the merge brought these in. It also looks like it matches with develop now so the changes are not found in "File changed"

**Review by digiwand — DISMISSED** [review]

**Review by blackdevelopa — DISMISSED** [review]

**Review by digiwand — DISMISSED** [review]

File: ui/pages/confirmations/components/confirm-page-container/index.scss [hunk 3]
```diff
@@ -2,9 +2,6 @@
 @import 'confirm-page-container-header/index';
 @import 'confirm-detail-row/index';
 @import 'confirm-page-container-navigation/index';
-///: BEGIN:ONLY_INCLUDE_IF(snaps)
```
Comment (hmalik88) [inline ui/pages/confirmations/components/confirm-page-container/index.scss:5]: why was this removed?

File: ui/components/app/app-components.scss [hunk 4]
```diff
@@ -104,6 +72,6 @@
 @import '../institutional/custody-confirm-link-modal/index';
 @import '../institutional/transaction-failed-modal/index';
 ///: END:ONLY_INCLUDE_IF
-///: BEGIN:ONLY_INCLUDE_IF(conf-redesign)
-@import 'confirm/header/header.scss';
+///: BEGIN:ONLY_INCLUDE_IF(snaps)
```
Comment (hmalik88) [inline ui/components/app/app-components.scss:75]: I'm not sure I follow what's going on here

File: ui/pages/confirmations/components/confirm-page-container/index.scss (hunk 3 above)
Comment (jpuri) [reply ui/pages/confirmations/components/confirm-page-container/index.scss:5]: @hmalik88 : `confirm-page-container` has now been moved to `pages/confirmations` as they are ownership of confirmations team and so I moved `snap-insight` to `/components/app/spans`. This is the reason for above change and also change in file [ui/components/app/app-components.scss](https://github.com/MetaMask/metamask-extension/pull/22639/files/6f6b9cf4936eaaed2d5190b40582109a9600b3c1#diff-210a3b6ffa98ce55e09288b37de7aa678ecf0e75015bc42786eee28f940552ba)

Comment (hmalik88) [reply ui/pages/confirmations/components/confirm-page-container/index.scss:5]: Gotcha, thanks!

**Review by hmalik88 — APPROVED** [review]
LGTM

**Review by blackdevelopa — APPROVED** [review]
